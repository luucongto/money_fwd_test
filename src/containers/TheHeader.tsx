import { CHeader, CToggler } from '@coreui/react'
import React from 'react'
import { useAppDispatch, useAppSelector } from '../redux/hooks'
import { selectSidebarShow, set } from '../redux/reducers/sidebar'

const TheHeader = (): JSX.Element => {
  const dispatch = useAppDispatch()
  const sidebarShow = useAppSelector(state => selectSidebarShow(state))

  const toggleSidebar = (): void => {
    const val = [true, 'responsive', undefined].includes(sidebarShow) ? false : 'responsive'
    dispatch(set(val))
  }

  return (
    <CHeader withSubheader>
      <CToggler
        inHeader
        className='ml-3 d-md-down-none'
        onClick={toggleSidebar}
      />
      <h4 className='card-title mb-0' style={{ alignSelf: 'center' }}>Account Management</h4>
    </CHeader>
  )
}

export default TheHeader
