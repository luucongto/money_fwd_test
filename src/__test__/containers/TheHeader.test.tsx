import React from 'react'
import userEvent from '@testing-library/user-event'
import TheHeader from '../../containers/TheHeader'
import { mount, ReactWrapper } from 'enzyme'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
// Before your test, do next setting
const middlewares = [] // If you are not using any middleware, leave the array empty or import it and put it in
const mockStore = configureStore(middlewares) // Setting of store based on the middlewares.
// Create the store to pass as prop in the <Provider>

describe('AccountDisplay display user', () => {
  let wrapperResult: ReactWrapper
  let store: any
  beforeEach(() => {
    const initState = {
      sidebar: {
        sidebarShow: 'responsive'
      }
    }

    store = mockStore(initState)
    wrapperResult = mount(
      <Provider store={store}>
        <TheHeader />
      </Provider>
    )
  })

  test('renders TheHeader without crashing', () => {
    expect(wrapperResult).toMatchSnapshot()
  })

  test('renders Account Management header', () => {
    expect(wrapperResult.text().includes('Account Management')).toBe(true)
  })

  test('toogleSidebar on header', () => {
    wrapperResult.find('.c-header-toggler').simulate('click')
    expect(store.getActions()).toEqual([{ type: 'sidebar/set', payload: false }])
  })
})
