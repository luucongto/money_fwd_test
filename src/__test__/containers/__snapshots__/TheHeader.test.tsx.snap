// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountDisplay display user renders TheHeader without crashing 1`] = `ReactWrapper {}`;

exports[`renders TheHeader without crashing 1`] = `ShallowWrapper {}`;
